src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [isTodosLoad, setIsTodosLoad] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setTodos(await getTodos());
+        setIsTodosLoad(true);
+      } catch (error) {
+        throw new Error('Something went wrong: Could load data');
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const checkStatus = useCallback((
+    todoCompleted: boolean,
+    filterStage: string,
+  ) => {
+    switch (filterStage) {
+      case 'completed':
+        return todoCompleted === true;
+      case 'active':
+        return todoCompleted === false;
+      default:
+        return true;
+    }
+  }, []);
```

Comment:
You don't need to wrap it in useCallback. You can move this function outside your component as helper